private/runtime/binding-spec.rkt

```diff
@@ -271,16 +280,24 @@
                ([spec specs])
        (simple-expand-internal spec st local-scopes))]
     
-    [(nest pv f inner-spec)
-     (define init-seq (get-pvar st pv))
+    [(nest depth pv f inner-spec)
+     (when (> depth 1)
+       (error "don't know how to handle depth > 1 yet"))
+
+     (define init-seq (if (= 0 depth)
+                          (list (get-pvar st pv))
+                          (get-pvar st pv)))
 
      (define res
        (start-nest f init-seq st inner-spec local-scopes))
      
      (match-define (nest-ret done-seq st^) res)
      
-     (set-pvar st^ pv done-seq)]
+     (set-pvar st^ pv (if (= 0 depth)
+                          (car done-seq)
+                          done-seq))]
```

Comment:
Currently it looks like the compiler generates nest-one rt forms in the depth = 0 case, so you shouldn't have to handle it in the runtime for nest. Alternatively, get rid of nest-one and keep and use the depth here. But not both.